.chloggen/upgrade-doublestar-pkg-stanza.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'enhancement'
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: 'pkg/stanza'
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: 'Upgrade version of doublestar from v3 to v4'
```

Comment:
Are there any breaking changes on v4 with respect to v3 that Collector users would face?

Reply:
https://github.com/bmatcuk/doublestar/blob/master/UPGRADING.md
Based on the notes above, v4 is a complete rewrite with a focus on performance. 
The breaking change is the `Glob()`. It now takes an `fs.FS` as its first argument.

I try to minimise changes by splitting the full path that includes the base path and pattern.
```diff
-matches, _ := doublestar.Glob(include)
+basepath, pattern := doublestar.SplitPattern(include)
+fsys := os.DirFS(basepath)
+matches, _ := doublestar.Glob(fsys, pattern)'
```
```Go
type Finder struct {
	Include []string `mapstructure:"include,omitempty"`
	Exclude []string `mapstructure:"exclude,omitempty"`
}
```
I am thinking that maybe can pass the filesystem path from the Finder struct, but that would break most of the API.

Does this answer your question?

Reply:
I don't believe this this change will not have any impact on users.

Reply:
Agreed, this seems to be Go API only, so the changelog is correct

Reply:
Thanks for looking into it @jackycsl :)